### PR TITLE
Add initial similarity API

### PR DESCRIPTION
**Pull request by ericholscher**
This ports over the logic from the CLI into a DRF API view

It's a basic start for now,
but gets at least a working example for local dev.

### CODE REVIEW

**Comment by ericholscher** [comment]
Next steps:

* One result per domain (fix a bunch of versions being returned at the same similarity)
* Only publishers w/ paid ads
* Limit to updated in the past 90 days

File: requirements/development.txt [hunk 1]
```diff
@@ -35,3 +35,7 @@ django_coverage_plugin==3.1.0
 
 # Used to build the ML model
 requests-cache==0.9.5
+
+# CORS headers
```
Comment (davidfischer) [inline requirements/development.txt:39]: These are required in production right? Probably want both of these in `base.txt` right?

File: config/settings/base.py [hunk 2]
```diff
@@ -538,3 +540,10 @@
     MIDDLEWARE.append(ADSERVER_IPADDRESS_MIDDLEWARE)
 if ADSERVER_GEOIP_MIDDLEWARE:
     MIDDLEWARE.append(ADSERVER_GEOIP_MIDDLEWARE)
+
+
+CORS_ALLOWED_ORIGINS = [
```
Comment (davidfischer) [inline config/settings/base.py:545]: This looks like a development setting. Perhaps make the prod setting read from env?

**Comment by ericholscher** [comment]
I pushed a couple changes to only return a single URL for a domain, and exclude unpaid publishers, which should make the results a bit cleaner to start.

File: adserver/analyzer/views.py [hunk 3]
```diff
@@ -1 +1,92 @@
...
+            urls = []
+            for url in unfiltered_urls:
+                domain = urlparse(url.url).netloc
+                if domain not in unique_domains:
+                    unique_domains.add(domain)
+                    urls.append(url)
+
+            if not len(urls) > 3:
+                return Response(
+                    {"error": "No similar URLs found"}, status=status.HTTP_404_NOT_FOUND
```
Comment (davidfischer) [inline adserver/analyzer/views.py:73]: I do think it's a little awkward to return HTML on success and JSON on failure but not a deal-breaker.

**Review by davidfischer — APPROVED** [review]
I think this is fine for an initial release and we'll refine after it goes out.

File: adserver/analyzer/views.py (hunk 3 above)
Comment (ericholscher) [reply adserver/analyzer/views.py:73]: Yea, there's a bunch of improvements here, just wanted to not 500 :)

File: config/settings/base.py [hunk 4]
```diff
@@ -538,3 +540,10 @@
...
+    "http://localhost:8000",
+    "http://127.0.0.1:8000",
+]
+CORS_ALLOW_HEADERS = ["*"]
```
[the remark this answers is not in the document]
Comment (ericholscher) [reply config/settings/base.py]: Bleh.. old comment that didn't submit.

**Pull request merged**